relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -1,116 +1,20 @@
-        match sampling_match {
-            Some(SamplingMatch {
-                sample_rate,
-                matched_rule_ids,
-                seed,
-            }) => {
-                let random_number = relay_sampling::evaluation::pseudo_random_from_uuid(seed);
-                relay_log::trace!(
-                    sample_rate,
-                    random_number,
-                    "applying dynamic sampling to matching event"
-                );
-
-                if random_number >= sample_rate {
-                    relay_log::trace!("dropping event that matched the configuration");
-                    SamplingResult::Drop(matched_rule_ids)
-                } else {
-                    relay_log::trace!("keeping event that matched the configuration");
-                    SamplingResult::Keep
-                }
-            }
-            None => {
-                relay_log::trace!("keeping event that didn't match the configuration");
-                SamplingResult::Keep
-            }
-        }
-    }
-}
-
-/// Gets the sampling match result by creating the merged configuration and matching it against
-/// the sampling configuration.
-fn get_sampling_match_result(
-    processing_enabled: bool,
-    project_state: Option<&ProjectState>,
-    root_project_state: Option<&ProjectState>,
-    dsc: Option<&DynamicSamplingContext>,
-    event: Option<&Event>,
-    now: DateTime<Utc>,
-) -> Option<SamplingMatch> {
-    // We want to extract the SamplingConfig from each project state.
-    let sampling_config = project_state.and_then(|state| state.config.dynamic_sampling.as_ref());
-    let root_sampling_config =
-        root_project_state.and_then(|state| state.config.dynamic_sampling.as_ref());
-
-    relay_sampling::evaluation::merge_configs_and_match(
-        processing_enabled,
-        sampling_config,
-        root_sampling_config,
-        dsc,
-        event,
-        now,
-    )
-}
-
-/// Runs dynamic sampling on an incoming event/dsc and returns whether or not the event should be
-/// kept or dropped.
-pub fn get_sampling_result(
-    processing_enabled: bool,
-    project_state: Option<&ProjectState>,
-    root_project_state: Option<&ProjectState>,
-    dsc: Option<&DynamicSamplingContext>,
-    event: Option<&Event>,
-) -> SamplingResult {
-    let sampling_result = get_sampling_match_result(
-        processing_enabled,
-        project_state,
-        root_project_state,
-        dsc,
-        event,
-        // For consistency reasons we take a snapshot in time and use that time across all code that
-        // requires it.
-        Utc::now(),
-    );
-    SamplingResult::determine_from_sampling_match(sampling_result)
-}
-
 /// Runs dynamic sampling if the dsc and root project state are not None and returns whether the
 /// transactions received with such dsc and project state would be kept or dropped by dynamic
 /// sampling.
 pub fn is_trace_fully_sampled(
```

Comment:
not a big fan of this function, but we agreed to keep it, maybe we can revisit it later